From d4d20f0d0c3592bdc736513bc594f6d0fa0f6a15 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Thu, 30 May 2019 09:54:57 +0300
Subject: [PATCH] Fix yocto build issues

- use QT_PLUGIN_PATH for qtsafelayouttool calls. Builds are using qt.conf
  (from OE_QMAKE_QTCONF_PATH env in yocto builds) and from QT_HOST_BINS
  directory in the nativesdk, which has target paths for QT_INSTALL_PLUGINS
  that causes qtsafelayouttool to fail to find any plugins.

- Use QT_HOST_BINS/get to point to the native sysroot

Change-Id: Ie042b7ef45c7a5e03b36008a2e2cbce1a9193139
---
 tools/featurespec/qtsaferenderer-tools.prf | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/tools/featurespec/qtsaferenderer-tools.prf b/tools/featurespec/qtsaferenderer-tools.prf
index b4daf86..773c8e5 100644
--- a/tools/featurespec/qtsaferenderer-tools.prf
+++ b/tools/featurespec/qtsaferenderer-tools.prf
@@ -1,3 +1,7 @@
+pluginpath.name = QT_PLUGIN_PATH
+pluginpath.value = $$[QT_HOST_LIBS/get]/plugins
+QT_TOOL_ENV += pluginpath
+
 defineTest(addSafeTargetEnv) {
     cmd = $$3
     $${1}_EXE = $$cmd
@@ -17,7 +21,7 @@ defineTest(addSafeTargetEnv) {
 defineTest(prepareSafeTool) {
     cmd = $$eval(QT_TOOL.$${2}.binary)
     isEmpty(cmd)|!exists($$cmd) {
-        cmd = $$[QT_HOST_BINS]/$${2}
+        cmd = $$[QT_HOST_BINS/get]/$${2}
         contains(QMAKE_HOST.os, Windows) {
             cmd = $${cmd}.exe
         }
@@ -44,7 +48,7 @@ prepareSafeTool(QMAKE_SAFELAYOUT, qtsafelayouttool)
     prepareSafeTool(QMAKE_SAFERCC, qtsafercctool)
     SAFERCC_FILE = $$QMAKE_SAFERCC_EXE
     isEmpty(SAFERCC_FILE)|!exists($$SAFERCC_FILE) {
-        cmd = $$absolute_path("rcc", $$[QT_HOST_BINS])
+        cmd = $$absolute_path("rcc", $$[QT_HOST_BINS/get])
         contains(QMAKE_HOST.os, Windows) {
             cmd = $$system_path($${cmd}.exe)
         }
